drivers/hud_apr/app/models/hud_apr/generators/shared/fy2024/question_four.rb

```diff
@@ -0,0 +1,101 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HudApr::Generators::Shared::Fy2024
+  class QuestionFour < Base
+    QUESTION_NUMBER = 'Question 4'.freeze
+
+    TABLE_HEADER = [
+      'Organization Name',
+      'Organization ID',
+      'Project Name',
+      'Project ID',
+      'HMIS Project Type',
+      'RRH Subtype',
+      'Coordinated Entry Access Point',
+      'Affiliated with a residential project',
+      'Project IDs of affiliations',
+      'CoC Number',
+      'Geocode',
+      'Victim Service Provider',
+      'HMIS Software Name',
+      'Report Start Date',
+      'Report End Date',
+      'Total Active Clients',
+      'Total Active Households',
+    ].freeze
+
+    HMIS_SOFTWARE_NAME = 'OpenPath HMIS Warehouse'.freeze
+
+    def self.table_descriptions
+      {
+        'Question 4' => 'HMIS Information',
+        'Q4a' => 'Project Identifiers in HMIS',
+      }.freeze
+    end
+
+    def detect_ce_participation(project)
+      # FIXME: this probably isn't right
+      # Column G should show the response to 2.09.1 effective as of the [report end date]. This element is a transactional data element, and only the most recent value for the report period is displayed.
+      records = project.ce_participations.filter do |record|
+        start = record.CEParticipationStatusStartDate || 100.years.ago
+        stop = record.CEParticipationStatusEndDate || 100.years.from_now
+        @report.start_date.between?(start, stop)
```

Comment:
```suggestion
        @report.end_date.between?(start, stop)
```
Otherwise I believe this is correct.  Basically find the most-recently changed record that overlapped the report end date.